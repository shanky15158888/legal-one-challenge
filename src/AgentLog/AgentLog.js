import React from 'react'
import Table from '../Table/Table';
import logs from '../json-data/logs.json'
import resolution from '../json-data/resolution.json'
import moment from 'moment';
import { Link } from 'react-router-dom';

export default function Agentlog(props) {
    const id = props.match && props.match.params && props.match.params.id ? props.match.params.id : ""
    const cols = ["Phone number", "Call date and time", "Resolution"]
    const tableData = [];

    const agentRecords = logs.filter((log) => {
        return log.agentIdentifier === id
    })

    const resolutionRecord = getResolution(agentRecords)

    resolutionRecord.forEach((item) => {
        const obj = {}
        obj['data'] = []
        obj.data.push({ 'data': item.number, type: "link", url: `/call/${item.number}` })
        obj.data.push({ 'data': getDateTimeFormat(item.dateTime) })
        obj.data.push({ 'data': item.resolution })
        tableData.push(obj)
    })

    return (
        <>
            <header><h2>Agent Logs</h2></header>
            <Link className="goRight" to="/">Home</Link>
            <Table cols={cols} rows={tableData} />
        </>
    )
}

const getResolution = (list) => {
    const arr = JSON.parse(JSON.stringify(list));
    const temp_arr = [];
    arr.forEach((item) => {
        resolution.forEach((subItem) => {
            if (item.identifier === subItem.identifier) {
                item['resolution'] = subItem.resolution
                temp_arr.push(item)
            }
        })
    })
    return temp_arr
}

const getDateTimeFormat = (date) => {
    return moment(date).format('DD/M/YYYY HH:MM:SS');
}