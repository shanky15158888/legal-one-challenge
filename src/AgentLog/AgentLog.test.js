import React from 'react';
import renderer from 'react-test-renderer';
import AgentLog from './AgentLog'
import { BrowserRouter as Router } from "react-router-dom";

test('AgentLog component testing', () => {
    const match = {
        params: {
            number: "356b03dc-9ec5-11e7-97a6-d501104f897e"
        }
    }
    const component = renderer.create(
        <Router>
            <AgentLog match={match} />
        </Router>,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});