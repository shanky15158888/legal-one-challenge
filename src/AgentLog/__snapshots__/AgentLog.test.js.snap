// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AgentLog component testing 1`] = `
Array [
  <header>
    <h2>
      Agent Logs
    </h2>
  </header>,
  <a
    className="goRight"
    href="/"
    onClick={[Function]}
  >
    Home
  </a>,
  <table
    className="table table-striped"
  >
    <thead>
      <tr>
        <th
          scope="col"
        >
          Phone number
        </th>
        <th
          scope="col"
        >
          Call date and time
        </th>
        <th
          scope="col"
        >
          Resolution
        </th>
      </tr>
    </thead>
    <tbody />
  </table>,
]
`;
